libp2p/protocols/pubsub/gossipsub.nim

```diff
@@ -360,6 +375,10 @@ method rpcHandler*(g: GossipSub,
 
       await handleData(g, t, msg.data)
 
+    # Don't send it to source peer, or peers that
+    # sent it during validation
+    toSendPeers.excl(seenPeers)
```

Comment:
it probably makes sense to exclude the "first" sending peer separately: this impl looks correct but in the majority of cases will unnecessarily allocate a hash set - when we do the sending peer separately, an empty hash set init will not use any memory

Reply:
Oh, didn't know that an empty set was free :) - done

Reply:
Actually I'm not sure it's free, initHashSet creates a seq. Maybe there is a way to create one without init.
But maybe a hashset is not necessary anyway, as a wise men once said: "`O` notation only matters when `n` is big, and `n` is generally small"
We could probably switch a seq for so few elements

Reply:
use `... = HashSet[...]()` - that allocates no memory - I'm not sure about the seq - there's a risk here that if validation is slow, N can grow quite a bit